packages/design-picker/src/components/index.tsx

```diff
@@ -115,11 +124,27 @@ const DesignPicker: React.FC< DesignPickerProps > = ( {
 	theme = 'light',
 	className,
 	highResThumbnails = false,
+	showCategoryFilter = false,
 } ) => {
+	const categories = gatherCategories( designs );
```

Comment:
It would be better to memo categories but maybe we can do it later.

Reply:
I had thought about this too, but [remembered this post](https://overreacted.io/before-you-memo/).
You're right we can do it later if necessary, but my guess is that the `<DesignPicker>` component doesn't render very often.

Reply:
Actually ... it's going to render every time the selection changes 🤔 maybe I'll see what `useMemo` is like, because `gatherCategories` is going to return a fresh array each time ...

Reply:
Just tried out `useMemo` in the profiler. It doesn't end up making a difference to the render. `<DesignPickerCategoryFilter>` still renders every time because the `selectedCategory` prop changes, so it doesn't matter whether the value of `categories` is identical or not. All we're saving is the computational work of merging the arrays, which is basically nothing compared to the work React is doing.